### PR TITLE
Fixes the CV parshift-order-2 gradient bug.

### PR DESCRIPTION
**Pull request by smite**
**Context:**

Fixes a bug in the CV order-2 parameter-shift method, where observables which are not descendants of the parametrized operation were not transformed at all during the partial derivative evaluation. They should instead be made to produce zeros.

**Description of the Change:**

* Replaces the expectation values of non-descendant observables with zeros.
* Adds a test for the bug.
* An existing test in `test_templates_subroutines.py` is changed so that it does not compare the par-shift Jacobian to a hard-coded array but to the finite-diff Jacobian instead.

### CODE REVIEW

File: pennylane/qnode.py [hunk 1]
```diff
@@ -1077,7 +1073,7 @@ def _pd_analytic(self, params, idx, force_order2=False, **kwargs):
 
                 # transform the observables
                 obs = [
-                    self._transform_observable(ob, ob_successors, w, Z)
+                    self._transform_observable(ob, w, Z)
```
Comment (smite) [inline pennylane/qnode.py]: For the old QNode I just implemented the simpler (but probably less efficient) solution, i.e. transforming all the observables. Z transforms all non-descendant observables to zero observables.

**Comment by smite** [comment]
What's the correct way to fix the failing `Interferometer` test?

File: pennylane/qnode.py (hunk 1 above)
Comment (smite) [reply pennylane/qnode.py]: This is assuming that the old QNode will go away at some point, of course.

File: pennylane/beta/qnodes/cv.py [hunk 2]
```diff
@@ -220,15 +220,17 @@ def _pd_analytic(self, idx, args, kwargs, **options):
...
-
-                # measure transformed observables
-                pd += self.evaluate_obs(obs, unshifted_args, kwargs)
+                obs = [self._transform_observable(x, w, Z) for x in desc]
+                # Measure the transformed observables.
+                # The other observables do not depend on this parameter instance,
+                # hence their partial derivatives are zero.
+                res = self.evaluate_obs(obs, unshifted_args, kwargs)
+
+                # add the measured pd:s to the correct locations
```
Comment (josh146) [inline pennylane/beta/qnodes/cv.py]: ?
```suggestion
                # add the measured pd's to the correct locations
```

File: pennylane/beta/qnodes/cv.py [hunk 3]
```diff
@@ -220,15 +220,17 @@ def _pd_analytic(self, idx, args, kwargs, **options):
...
-                    for x in self.circuit.observables
-                ]
-
-                # measure transformed observables
-                pd += self.evaluate_obs(obs, unshifted_args, kwargs)
+                obs = [self._transform_observable(x, w, Z) for x in desc]
+                # Measure the transformed observables.
+                # The other observables do not depend on this parameter instance,
+                # hence their partial derivatives are zero.
+                res = self.evaluate_obs(obs, unshifted_args, kwargs)
```
Comment (josh146) [inline pennylane/beta/qnodes/cv.py]: Ah, so the main difference is removing the `if x in desc else x`, and replacing all other observable PD's with 0?

File: tests/beta/test_qnode_cv.py [hunk 4]
```diff
@@ -226,27 +209,9 @@ def qf(x, y):
...
-        q = CVQNode(qf, dev)
-
-        par = [0.543, 0.123, 0.654, -0.629]
-
-        grad_A = q.jacobian(par, method="A")
-        grad_F = q.jacobian(par, method="F")
-
-        # the different methods agree
-        assert grad_A == pytest.approx(grad_F, abs=tol)
-
```
Comment (josh146) [inline tests/beta/test_qnode_cv.py]: How come these two tests were deleted?

File: tests/beta/test_qnode_cv.py [hunk 5]
```diff
@@ -316,22 +282,38 @@ def circuit(x):
...
+            qml.Squeezing(r1, phi1, wires=[0])
+            return qml.expval(qml.NumberOperator(0))
+
+        q = CVQNode(qf, dev)
+        par = [0.543, 0.123, 0.654, -0.629]
+
+        grad_A = q.jacobian(par, method="A")
+        grad_F = q.jacobian(par, method="F")
+        # the different methods agree
+        assert grad_A == pytest.approx(grad_F, abs=tol)
```
Comment (josh146) [inline tests/beta/test_qnode_cv.py]: Oh, they've just been moved? Were there any other changes?

File: tests/beta/test_qnode_cv.py [hunk 6]
```diff
@@ -460,6 +440,66 @@ def circuit(x):
...
+
+        res = node(par)
+        res_true = np.sinh(np.abs(par)) ** 2  # analytical result
+        assert res == pytest.approx(res_true, abs=tol)
+
+        grad_A = node.jacobian([par], method="A")
+        grad_F = node.jacobian([par], method="F")
+        grad_true = np.diag(np.sinh(2 * np.abs(par)) * np.sign(par))  # analytical gradient
+        assert grad_A == pytest.approx(grad_F, abs=tol)
+        assert grad_A == pytest.approx(grad_true, abs=tol)
```
Comment (josh146) [inline tests/beta/test_qnode_cv.py]: :+1:

**Pull request merged**